File: src/main/java/me/lunaluna/fabric/elytrarecast/Timer.java
package me.lunaluna.fabric.elytrarecast;

import java.util.function.Supplier;

public class Timer {

    private final Supplier<Integer> getCooldown;
    private long previousTime = 0L;

    public Timer(Supplier<Integer> getCooldown) {
        this.getCooldown = getCooldown;
    }

    public void runOnCooldown(Runnable runnable) {
        long time = System.currentTimeMillis();
        long diff = time - previousTime;
        if (diff >= getCooldown.get()) {
            runnable.run();
            previousTime = time;
        }
    }
}

File: src/main/java/me/lunaluna/fabric/elytrarecast/Timer.kt
package me.lunaluna.fabric.elytrarecast

class Timer(val getCooldown: () -> Int) {


    var previousTime: Long = 0L
    private val cooldown get() = getCooldown()

    fun runOnCooldown(runnable: () -> Unit) {
        val time = System.currentTimeMillis()
        val diff = time - previousTime
        if (diff >= cooldown) { runnable() }
        previousTime = time
    }
}